src/basic/capability-util.c

```diff
@@ -295,7 +296,7 @@ int drop_privileges(uid_t uid, gid_t gid, uint64_t keep_capabilities) {
         if (setresgid(gid, gid, gid) < 0)
                 return log_error_errno(errno, "Failed to change group ID: %m");
 
-        if (setgroups(0, NULL) < 0)
+        if (maybe_setgroups(0, NULL) < 0)
```

Comment:
```sh
$ git grep drop_priv src/ | grep -v test
src/basic/capability-util.c:int drop_privileges(uid_t uid, gid_t gid, uint64_t keep_capabilities) {
src/basic/capability-util.h:int drop_privileges(uid_t uid, gid_t gid, uint64_t keep_capabilities);
src/coredump/coredump.c:        return drop_privileges(uid, gid, 0);
src/network/networkd.c:        r = drop_privileges(uid, gid,
src/resolve/resolved.c:        r = drop_privileges(uid, gid,
src/timesync/timesyncd.c:        r = drop_privileges(uid, gid, (1ULL << CAP_SYS_TIME));
```
Seems like you can't run all these binaries inside the unprivileged user namespace: `setresuid(uid, uid, uid)` fails. Why do we need this?